yjit/src/backend/ir.rs

```diff
@@ -1094,9 +1097,13 @@ impl Assembler
     {
         #[cfg(feature = "disasm")]
         let start_addr = cb.get_write_ptr().raw_ptr();
+        let old_page_fault = cb.page_fault;
 
         let alloc_regs = Self::get_alloc_regs();
-        let gc_offsets = self.compile_with_regs(cb, alloc_regs);
+        let mut gc_offsets = self.clone().compile_with_regs(cb, alloc_regs.clone());
```

Comment:
Hmmm. Not loving this because having to `clone()` the assembler so we can retry seems expensive 😬 

I'm wondering if we could try to predict (and maybe slightly overestimate) how much code we'll generate, and pre-emptively move to the next page.

Otherwise we could try to make compile_with_regs not be destructive so that we can re-run it without cloning.

On the other hand I understand if this is just for prototyping, but in that case we should add a comment now to explain the tradeoff.

Reply:
I implemented this at `compile_with_regs` to share it between Intel and Arm, but if we do that inside it, I think we could call `x86_emit` or `arm64_emit` twice without cloning it because it doesn't move `self`.

I thought about prediction as well, but it seems like it would end with up a lengthy code or unnecessarily large gaps. So I'll try the above idea first.

Reply:
I see what you mean. Your idea of calling x86_emit or arm64_emit twice to avoid cloning makes a lot of sense 🙂